core/xchaincore.go

```diff
@@ -503,16 +506,18 @@ func (xc *XChainCore) doMiner() {
 	utxovmLastID := xc.Utxovm.GetLatestBlockid()
 
 	// 如果Walk一直失败，建议不要挖矿了，而是报警处理
-	for !bytes.Equal(ledgerLastID, utxovmLastID) {
-		xc.log.Warn("ledger last blockid is not equal utxovm last id")
-		err := xc.Utxovm.Walk(ledgerLastID)
-		if err != nil {
-			xc.log.Error("Walk error ", "ledger blockid", global.F(ledgerLastID),
-				"utxo blockid", global.F(utxovmLastID))
-			return
+	if xc.walkLoop {
+		for !bytes.Equal(ledgerLastID, utxovmLastID) {
+			xc.log.Warn("ledger last blockid is not equal utxovm last id")
```

Comment:
(1) Here,  we should call "Walk" at least  once.  But, your code will cause Walk not be called if the flag is false
(2) the option flag 'walkLoop' can be changed to another word, such as "failover". Or, "failfast" in opposite.